File: source/clibCoreCs/addonCollection_core/processAddonStyleEditor.cs
﻿using System;

//========================================================================
// This conversion was produced by the Free Edition of
// Instant C# courtesy of Tangible Software Solutions.
// Order the Premium Edition at https://www.tangiblesoftwaresolutions.com
//========================================================================

using Contensive.BaseClasses;
using Contensive.Core;
using Models.Entity;
using Contensive.Core.Controllers;

//
namespace Contensive.Addons.Core
{
	public class processAddonStyleEditorClass : Contensive.BaseClasses.AddonBaseClass
	{
		//
		//====================================================================================================
		/// <summary>
		/// getFieldEditorPreference remote method
		/// </summary>
		/// <param name="cp"></param>
		/// <returns></returns>
		public override object Execute(Contensive.BaseClasses.CPBaseClass cp)
		{
			string result = "";
			try
			{
				CPClass processor = (CPClass)cp;
				coreClass cpCore = processor.core;
				//
				// save custom styles
				if (cpCore.doc.authContext.isAuthenticated & cpCore.doc.authContext.isAuthenticatedAdmin(cpCore))
				{
					int addonId = cpCore.docProperties.getInteger("AddonID");
					if (addonId > 0)
					{
						Models.Entity.addonModel styleAddon = Models.Entity.addonModel.create(cpCore, addonId);
						if (styleAddon.StylesFilename.content != cpCore.docProperties.getText("CustomStyles"))
						{
							styleAddon.StylesFilename.content = cpCore.docProperties.getText("CustomStyles");
							styleAddon.save(cpCore);
							//
							// Clear Caches
							//
							cpCore.cache.invalidateAllObjectsInContent(addonModel.contentName);
						}
					}
				}
			}
			catch (Exception ex)
			{
				cp.Site.ErrorReport(ex);
			}
			return result;
		}
	}
}

File: etc/clibCoreVb/addonCollection_core/processAddonStyleEditor.vb

Option Explicit On
Option Strict On

Imports Contensive.BaseClasses
Imports Contensive.Core
Imports Contensive.Core.Models.Entity
Imports Contensive.Core.Controllers
Imports Contensive.Core.Controllers.genericController
'
Namespace Contensive.Addons.Core
    Public Class processAddonStyleEditorClass
        Inherits Contensive.BaseClasses.AddonBaseClass
        '
        '====================================================================================================
        ''' <summary>
        ''' getFieldEditorPreference remote method
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <returns></returns>
        Public Overrides Function execute(cp As Contensive.BaseClasses.CPBaseClass) As Object
            Dim result As String = ""
            Try
                Dim processor As CPClass = DirectCast(cp, CPClass)
                Dim cpCore As coreClass = processor.core
                '
                ' save custom styles
                If cpCore.doc.authContext.isAuthenticated() And cpCore.doc.authContext.isAuthenticatedAdmin(cpCore) Then
                    Dim addonId As Integer
                    addonId = cpCore.docProperties.getInteger("AddonID")
                    If (addonId > 0) Then
                        Dim styleAddon As addonModel = addonModel.create(cpCore, addonId)
                        If (styleAddon.StylesFilename.content <> cpCore.docProperties.getText("CustomStyles")) Then
                            styleAddon.StylesFilename.content = cpCore.docProperties.getText("CustomStyles")
                            styleAddon.save(cpCore)
                            '
                            ' Clear Caches
                            '
                            Call cpCore.cache.invalidateAllObjectsInContent(addonModel.contentName)
                        End If
                    End If
                End If
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
            End Try
            Return result
        End Function
    End Class
End Namespace
